cot/src/openapi.rs

```diff
@@ -0,0 +1,1179 @@
+    }
+
+    Inner(handler, PhantomData, PhantomData)
+}
+
+pub(crate) trait BoxApiEndpointRequestHandler: BoxRequestHandler + AsApiRoute {
+    // TODO: consider removing this when Rust trait_upcasting is stabilized and we
+    // bump the MSRV (lands in Rust 1.86)
+    fn as_box_request_handler(&self) -> &(dyn BoxRequestHandler + Send + Sync);
+}
+
+pub(crate) fn into_box_api_endpoint_request_handler<HandlerParams, H>(
+    handler: H,
+) -> impl BoxApiEndpointRequestHandler
+where
+    H: RequestHandler<HandlerParams> + AsApiRoute + Send + Sync,
+{
+    struct Inner<HandlerParams, H>(H, PhantomData<fn() -> HandlerParams>);
+
+    impl<HandlerParams, H> BoxRequestHandler for Inner<HandlerParams, H>
+    where
+        H: RequestHandler<HandlerParams> + AsApiRoute + Send + Sync,
+    {
+        fn handle(
+            &self,
+            request: Request,
+        ) -> Pin<Box<dyn Future<Output = cot::Result<Response>> + Send + '_>> {
+            Box::pin(self.0.handle(request))
+        }
+    }
+
+    impl<HandlerParams, H> AsApiRoute for Inner<HandlerParams, H>
+    where
+        H: RequestHandler<HandlerParams> + AsApiRoute + Send + Sync,
+    {
+        fn as_api_route(
+            &self,
+            route_context: &RouteContext<'_>,
+            schema_generator: &mut SchemaGenerator,
+        ) -> PathItem {
+            self.0.as_api_route(route_context, schema_generator)
+        }
+    }
+
+    impl<HandlerParams, H> BoxApiEndpointRequestHandler for Inner<HandlerParams, H>
+    where
+        H: RequestHandler<HandlerParams> + AsApiRoute + Send + Sync,
+    {
+        fn as_box_request_handler(&self) -> &(dyn BoxRequestHandler + Send + Sync) {
+            self
+        }
+    }
+
+    Inner(handler, PhantomData)
+}
+
+/// A wrapper type that allows using non-OpenAPI handlers and request parameters
+/// in OpenAPI routes.
+///
+/// If you need an extractor or a handler that does not implement
+/// [`AsApiOperation`]/[`ApiOperationPart`], you can wrap it in a `NoApi` to
+/// indicate that it should just be ignored during OpenAPI generation.
+///
+/// # Examples
+///
+/// ```
+/// use cot::openapi::NoApi;
+/// use cot::request::extractors::FromRequestParts;
+/// use cot::response::Response;
+/// use cot::router::Route;
+/// use cot::router::method::openapi::api_get;
+/// use http::request::Parts;
+///
+/// struct MyExtractor;
+/// impl FromRequestParts for MyExtractor {
+///     async fn from_request_parts(parts: &mut Parts) -> cot::Result<Self> {
+///         // ...
+/// #         unimplemented!()
+///     }
+/// }
+///
+/// async fn handler(NoApi(extractor): NoApi<MyExtractor>) -> cot::Result<Response> {
+///     // MyExtractor doesn't have to implement ApiOperationPart and
+///     // doesn't show up in the OpenAPI spec
+/// #     unimplemented!()
+/// }
+///
+/// let router = cot::router::Router::with_urls([Route::with_api_handler(
+///     "/with_api",
+///     // POST will be ignored in OpenAPI spec
+///     api_get(handler),
```

Comment:
I don't get it, we don't specify `.post(handler)`, so would `POST` not appear anyway? What difference `NoApi` does here?

Reply:
I think it was just a copy-and-paste error, I just removed it.